### ISSUE TITLE
TypeError: [object Object] is not a function, it is object

### ISSUE THREAD

**Issue by eximius313** [sbt/sbt-less#80]
I have two environments: Windows 7 and GNU/Linux Debian Wheezy
I'm using Play Framework with sbt-less library.

On Windows machine my .less file compiles and runs just fine.
On Linux machine if I don't have pure Node.js installed and I'm using Trireme, I get this exception:

```
[info] LESS compiling on 1 source(s)
[error] TypeError: [object Object] is not a function, it is object. (/home/users/project/project/project/target/node-modules/webjars/less/lib/less/index.js#4)    at /home/users/project/project/pr
oject/target/node-modules/webjars/less/lib/less/index.js:4
[error]     at /home/users/project/project/project/target/node-modules/webjars/less/lib/less-node/index.js:1
[error]     at module.js:456
[error]     at module.js:474
[error]     at module.js:356
[error]     at module.js:312
[error]     at module.js:364
[error]     at require (module.js:380)
[error]     at /home/users/project/project/project/target/node-modules/webjars/less/index.js:1
[error]     at module.js:456
[error]     at module.js:474
[error]     at module.js:356
[error]     at module.js:312
[error]     at module.js:364
[error]     at require (module.js:380)
[error]     at /home/users/project/project/project/target/less/sbt-less-1.1.0.jar:14
[error]     at /home/users/project/project/project/target/less/sbt-less-1.1.0.jar:3
[error]     at module.js:456
[error]     at module.js:474
[error]     at module.js:356
[error]     at module.js:312
[error]     at module.js:497
[error]     at startup (trireme.js:142)
[error]     at trireme.js:923
```

With Node.js error is slightly different:

```
[info] LESS compiling on 1 source(s)
[error]
[error] /home/users/project/project/project/target/node-modules/webjars/less/lib/less/index.js:13
[error]         functions: require('./functions')(environment),
[error]                                          ^
[error] TypeError: object is not a function
[error]     at module.exports (/home/users/project/project/project/target/node-modules/webjars/less/lib/less/index.js:13:42)
[error]     at Object.<anonymous> (/home/users/project/project/project/target/node-modules/webjars/less/lib/less-node/index.js:5:12)
[error]     at Module._compile (module.js:456:26)
[error]     at Object.Module._extensions..js (module.js:474:10)
[error]     at Module.load (module.js:356:32)
[error]     at Function.Module._load (module.js:312:12)
[error]     at Module.require (module.js:364:17)
[error]     at require (module.js:380:17)
[error]     at Object.<anonymous> (/home/users/project/project/project/target/node-modules/webjars/less/index.js:1:80)
[error]     at Module._compile (module.js:456:26)
```

Could you please fix that?
Thank you in advance

**Comment by eximius313** [comment]
Ok, I have finally found a solution - one need to delete two folders:
```project/target``` and ```project/project``` as stated here: https://github.com/sbt/sbt-less/issues/68
Could you please do something about it? Because as I can see that a lot of people have this issue...

**Comment by adis-me** [comment]
I had the same error when upgrading to Play 2.5.0. Deleting `project/target` and 'project/project' solved the issue.

**Comment by eximius313** [comment]
Again the same problem when upgrading to Play 2.5.0:
```
module.js:328
[error]     throw err;
[error]     ^
[error] 
[error] Error: Cannot find module 'D:\Projects\myproject\target\less\lessc.js'
[error]     at Function.Module._resolveFilename (module.js:326:15)
[error]     at Function.Module._load (module.js:277:25)
[error]     at Function.Module.runMain (module.js:430:10)
[error]     at startup (node.js:141:18)
[error]     at node.js:980:3
[error] module.js:328
[error]     throw err;
[error] module.js:328
[error]     throw err;
[error]     ^
[error] 
[error] Error: Cannot find module 'D:\Projects\myproject\target\jshint\jshint-shell.js'
[error]     at Function.Module._resolveFilename (module.js:326:15)
[error]     at Function.Module._load (module.js:277:25)
[error]     at Function.Module.runMain (module.js:430:10)
[error]     at startup (node.js:141:18)
[error]     at node.js:980:3
```
solved by deleting ```project/target``` and ```project/project```

**Comment by mikhalchenko-alexander** [comment]
Similar error in Play 2.5.4. Deleting ```project/target``` and ```project/project``` didn't help.
```
[error] /Users/aam/Documents/idea_projects/play-bootstrap/target/less/lessc.js:148
[error]                 .then(handleResult, handleLessError);
[error]                  ^
[error] 
[error] TypeError: less.render(...).then is not a function
[error]     at /Users/aam/Documents/idea_projects/play-bootstrap/target/less/lessc.js:148:18
[error]     at tryToString (fs.js:449:3)
[error]     at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:436:12)
```

**Comment by pme123** [comment]
exactly same problem here with Play 2.5.3

**Comment by jroper** [comment]
@mikhalchenko-alexander that's not a similar error, that's a completely different error.

The original error reported in this issue is a duplicate of #68. The problem is that the webjars extractor cache is not deleting old files. Where this is a problem is lets say the old webjar had a file like `foo.js`, but then in the new webjar, `foo.js` had been split into many files, so it was replaced with a directory called `foo` which had an `index.js` file that did a `require` on each file in the directory. The problem is that the existence of the old `foo.js` will mean that `require` will load that, and not `foo/index.js`. And that's exactly what's happening here with less.

**Comment by jroper** [comment]
This is fixed in #161. Note that once upgraded to a version of sbt-web that includes this fix, if you have already experienced the problem, you may still need to deleted your `project/target` folder, since that fix is able to ensure that your project will never get into this state where this error happens, but it doesn't actually fix the problem if it already happened previously with an old version of sbt-web.

**Comment by jroper** [comment]
sbt-web 1.4.3 is released with this fix, but do read the comment above as merely upgrading won't fix your project if it's already in this state where you get this error, you have to delete your `project/target` directory, and then the upgrade will prevent you from getting into this state with future upgrades of the less plugin.
